azurerm/internal/services/eventgrid/tests/eventgrid_domain_topic_resource_test.go

```diff
@@ -0,0 +1,149 @@
+			},
+		},
+	})
+}
+
+func testCheckAzureRMEventGridDomainTopicDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).EventGrid.DomainTopicsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_eventgrid_domain_topic" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		domainName := rs.Primary.Attributes["domain_name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := client.Get(ctx, resourceGroup, domainName, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("EventGrid Domain Topic still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMEventGridDomainTopicExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).EventGrid.DomainTopicsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		domainName := rs.Primary.Attributes["domain_name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for EventGrid Domain Topic: %s", name)
+		}
+
+		resp, err := client.Get(ctx, resourceGroup, domainName, name)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: EventGrid Domain Topic %q (resource group: %s) does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on EventGrid.DomainTopicsClient: %s", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMEventGridDomainTopic_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+resource "azurerm_eventgrid_domain" "test" {
+  name                = "acctestegdomain-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+resource "azurerm_eventgrid_domain_topic" "test" {
+  name                = "acctestegtopic-%d"
+  domain_name         = azurerm_eventgrid_domain.test.name
+  resource_group_name = azurerm_resource_group.test.name
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger)
+}
+
+func testAccAzureRMEventGridDomainTopic_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMEventGridDomain_basic(data)
```

Comment:
i think this should be
```suggestion
	template := testAccAzureRMEventGridDomainTopic_basic(data)
```